File: enode/src/main/java/org/enodeframework/eventing/EventProcessContext.java
package org.enodeframework.eventing;

import java.util.concurrent.CompletableFuture;

/**
 * Represents the event processing context.
 */
public interface EventProcessContext {
    /**
     * Notify the event has been processed.
     */
    CompletableFuture<Boolean> notifyEventProcessed();
}

File: enode/src/main/java/org/enodeframework/eventing/EventProcessContext.kt
package org.enodeframework.eventing

import java.util.concurrent.CompletableFuture

/**
 * Represents the event processing context.
 */
interface EventProcessContext {
    /**
     * Notify the event has been processed.
     */
    fun notifyEventProcessed(): CompletableFuture<Boolean>
}
